File: w02/d02/sean_talia/cars/lib/car.rb
class Car

  attr_reader :horsepower, :fuel

  def initialize(horsepower, fuel)
    @horsepower = horsepower
    if fuel > 5
      @fuel = 5
    elsif fuel < 0
      @fuel = 0
    else
      @fuel = fuel
    end
  end

  def rev()
    "Vroom vroom!"
  end

  def drive()
    if fuel > 0
      @fuel -= 1
      true
    else
      false
    end
  end

end

File: w10/d04/najee_gardner/car.js
// var carFunctions = {
//   drive: function() {
//     this.fuel--;
//     return 'Vroom!'
//   },
//
//   refuel: function() {
//     this.fuel = 100;
//   }
// };

function Car(model, color) {
  this.model = model;
  this.color = color;
  this.fuel = 100;

  // this.__proto__ = carFunctions;
};
Car.prototype.drive = function() {
  this.fuel--;
  return 'Vroom!';
}
Car.prototype.refuel = function() {
  this.fuel = 100;
}

// function carFactory(model, color) {
//   var car = new Car(model, color);
//   return car;
// }
//
// var evo = carFactory('Mitsubishi Lance Evo IX', 'nightblue');
